src/Mvc/Mvc.ApiExplorer/test/EndpointMetadataApiDescriptionProviderTest.cs

```diff
@@ -329,16 +330,21 @@ static void AssertPathParameter(ApiDescription apiDescription)
     [Fact]
     public void AddsFromQueryParameterAsQuery()
     {
-        static void AssertQueryParameter(ApiDescription apiDescription)
+        static void AssertQueryParameter<T>(ApiDescription apiDescription)
         {
             var param = Assert.Single(apiDescription.ParameterDescriptions);
-            Assert.Equal(typeof(int), param.Type);
-            Assert.Equal(typeof(int), param.ModelMetadata.ModelType);
+            Assert.Equal(typeof(T), param.Type);
+            Assert.Equal(typeof(T), param.ModelMetadata.ModelType);
             Assert.Equal(BindingSource.Query, param.Source);
         }
 
-        AssertQueryParameter(GetApiDescription((int foo) => { }, "/"));
-        AssertQueryParameter(GetApiDescription(([FromQuery] int foo) => { }));
+        AssertQueryParameter<int>(GetApiDescription((int foo) => { }, "/"));
+        AssertQueryParameter<int>(GetApiDescription(([FromQuery] int foo) => { }));
+        AssertQueryParameter<TryParseStringRecordStruct>(GetApiDescription(([FromQuery] TryParseStringRecordStruct foo) => { }));
+        AssertQueryParameter<int[]>(GetApiDescription((int[] foo) => { }, "/"));
+        AssertQueryParameter<string[]>(GetApiDescription((string[] foo) => { }, "/"));
+        AssertQueryParameter<StringValues>(GetApiDescription((StringValues foo) => { }, "/"));
+        AssertQueryParameter<TryParseStringRecordStruct[]>(GetApiDescription((TryParseStringRecordStruct[] foo) => { }, "/"));
```

Comment:
GetApiDescription defaults to "GET", can we add a test case for the non-ShouldDisableInferredBody()-case?